### PR TITLE
TST: add test to io/formats/test_to_html.py to close GH6131

### LINKED ISSUE #6131: Missing ellipsis cells in pivot_table.to_html when limited by max_cols and len(rows) > 1
``` python
import pandas
df = pandas.DataFrame([
    {'a': 'aa', 'b': 'ba', 'c': 'ca', 'd': 1},
    {'a': 'ab', 'b': 'bb', 'c': 'cc', 'd': 2},
    {'a': 'ac', 'b': 'bc', 'c': 'cc', 'd': 3}
])

df.pivot_table(cols=['a'], rows=['b', 'c']).to_html(max_cols=2)
```

This renders like this:
broken_output

while this:

``` python
df.pivot_table(cols=['a'], rows=['b']).to_html(max_cols=2)
```

renders like this:
slection_656

As you can see in the first table, there are no ellipsis cells like in the second one.
I inspected the code to find the reason for this bug but couldn't find…

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [x] closes #6131
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@simonjayhawkins : Thanks for this!  Could you rebase onto `master` and run the CI again?

**Comment by simonjayhawkins** [comment]
@gfyoung 

The rebase failed the CI checks again. I'll wait until #22597 is merged into master before trying again.

File: pandas/tests/io/formats/test_to_html.py
Comment (WillAyd) [inline pandas/tests/io/formats/test_to_html.py]: Rather than creating `df` and then pivoting can you simplify and just create the desired frame (i.e. the result of the pivot)?

File: pandas/tests/io/formats/test_to_html.py
Comment (WillAyd) [inline pandas/tests/io/formats/test_to_html.py]: Minor nit but don't need full link - can just do `# GH 6131`

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by simonjayhawkins** [comment]
@WillAyd i made a couple of additional changes to your request in the latest commit which on hindsight may not have been such a good idea.  i changed the name of the test since it no longer includes a pivot table command and secondly i took the opportunity of fixing a typo in the original problem description and so now the test does not faithfully recreate the original issue. will be happy to amend as necessary.

**Review by WillAyd — APPROVED** [review]
Thanks this lgtm - @mroeschke what do you think?

**Review by mroeschke — APPROVED** [review]
LGTM as well

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**
